pkg/resource/training_job/custom_set_output.go

```diff
@@ -17,75 +17,50 @@
+	svccommon "github.com/aws-controllers-k8s/sagemaker-controller/pkg/common"
 	"github.com/aws/aws-sdk-go/aws"
 	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
-	corev1 "k8s.io/api/core/v1"
 )
 
-// customDescribeTrainingJobSetOutput sets the resource ResourceSynced condition to False if
-// TrainingJob is being modified by AWS. It has an additional check on the debugger status.
-func (rm *resourceManager) customDescribeTrainingJobSetOutput(
-	ctx context.Context,
-	r *resource,
-	resp *svcsdk.DescribeTrainingJobOutput,
-	ko *svcapitypes.TrainingJob,
-) (*svcapitypes.TrainingJob, error) {
-	trainingJobStatus := resp.TrainingJobStatus
-	debuggerRuleInProgress := false
-	if resp.DebugRuleEvaluationStatuses != nil {
-		for _, rule := range resp.DebugRuleEvaluationStatuses {
-			if rule.RuleEvaluationStatus != nil && *rule.RuleEvaluationStatus == svcsdk.RuleEvaluationStatusInProgress {
-				debuggerRuleInProgress = true
-				rm.customSetOutput(r, aws.String(svcsdk.TrainingJobStatusInProgress), ko)
-				break
-			}
-		}
+var (
+	trainingJobModifyingStatuses = []string{
+		svcsdk.TrainingJobStatusInProgress,
+		svcsdk.TrainingJobStatusStopping,
 	}
-
-	if !debuggerRuleInProgress {
-		rm.customSetOutput(r, trainingJobStatus, ko)
+	ruleModifyingStatuses = []string{
+		svcsdk.RuleEvaluationStatusInProgress,
+		svcsdk.RuleEvaluationStatusStopping,
 	}
+	resourceName = resourceGK.Kind
+)
 
-	return ko, nil
-}
-
-// customSetOutput sets ConditionTypeResourceSynced condition to True or False
-// based on the trainingJobStatus on AWS so the reconciler can determine if a
-// requeue is needed
-func (rm *resourceManager) customSetOutput(
-	r *resource,
-	trainingJobStatus *string,
-	ko *svcapitypes.TrainingJob,
-) {
-	if trainingJobStatus == nil {
+// customDescribeTrainingJobSetOutput sets the resource ResourceSynced condition to False if
+// TrainingJob is being modified by AWS. It checks for debug and profiler rule status in addition to TrainingJobStatus
+func (rm *resourceManager) customDescribeTrainingJobSetOutput(r *resource) {
+	trainingJobStatus := r.ko.Status.TrainingJobStatus
+	// early exit if training job is InProgress
+	if trainingJobStatus != nil && *trainingJobStatus == svcsdk.TrainingJobStatusInProgress {
+		svccommon.SetSyncedCondition(r, trainingJobStatus, &resourceName, &trainingJobModifyingStatuses)
 		return
 	}
 
-	syncConditionStatus := corev1.ConditionUnknown
-	if *trainingJobStatus == svcsdk.TrainingJobStatusCompleted || *trainingJobStatus == svcsdk.TrainingJobStatusStopped || *trainingJobStatus == svcsdk.TrainingJobStatusFailed {
-		syncConditionStatus = corev1.ConditionTrue
-	} else {
-		syncConditionStatus = corev1.ConditionFalse
-	}
-
-	var resourceSyncedCondition *ackv1alpha1.Condition = nil
-	for _, condition := range ko.Status.Conditions {
-		if condition.Type == ackv1alpha1.ConditionTypeResourceSynced {
-			resourceSyncedCondition = condition
-			break
+	for _, rule := range r.ko.Status.DebugRuleEvaluationStatuses {
```

Comment:
are debugger and profiler completely independent, isolated jobs - as in can they be run both with and without the other ?

Reply:
yes